### PR TITLE
[AIRFLOW-6829] Auto-apply apply_default

### PR DESCRIPTION
**Pull request by mik-laj**
This is a continuation of my previous PR, but I migrate it to my private account (mik-laj vs PolideaInternal).
https://github.com/apache/airflow/pull/7450

I did a rebase so we can merge this change. Now there are no compatibility issues for backport providers anymore, but if we merge this we will have to make sure proviider is only used with a compatible version of Airflow.   All user operators are still compatible.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
What does it mean 'sompatible version" ? will the change still work with 2.0.0 ?

**Comment by mik-laj** [comment]
>  'compatible version"

Any version that doesn't require the apply_defaults decorator, so any version that includes changes from that PR.

**Comment by ashb** [comment]
This leads to any interesting question we haven't asked or answered yet: how far back do we want provider modules to continue working for?

Right not they will work on 2.0.0+, bit when this is merged then the min Airflow version becomes (for example) 2.1.0.

(Let's not go with any automatic rewriting anymore! Not sure it would really be possible here, release wise anyway, changing the code would be fine I'm sure)

**Comment by potiuk** [comment]
But will the 'auto apply' not work for 2.0.0? I had the impression that
this change was backwards compatible.

And yes. Refactoring only made sense for 1.10

sob., 27 mar 2021, 19:35 użytkownik Ash Berlin-Taylor <
***@***.***> napisał:

**Comment by mik-laj** [comment]
>  I had the impression that this change was backwards compatible.

It's backwards compatible, because all olds operators are still workings.  Only operators that do not have a decorator do not work with older versions. And for PR, I update all operators so that they don't have a decorator.

File: airflow/providers/google/cloud/operators/cloud_sql.py [hunk 1]
```diff
@@ -491,26 +487,6 @@ class CloudSQLDeleteInstanceOperator(CloudSQLBaseOperator):
     )
     # [END gcp_sql_delete_template_fields]
 
-    @apply_defaults
```
Comment (mik-laj) [inline airflow/providers/google/cloud/operators/cloud_sql.py:494]:
```
airflow/providers/google/cloud/operators/cloud_sql.py:490:4: W0235: Useless super delegation in method '__init__' (useless-super-delegation)
```

**Comment by mik-laj** [comment]
I will add this to the Airflow 2.1 milestone so that we remember to talk about it, as I don't have the capacity to start a mailing list discussion right now.

**Comment by mik-laj** [comment]
I sent message to mailing list: https://lists.apache.org/thread.html/r713f180120d0a39b53567812eb5db34f992ec81979818d2175598b71%40%3Cdev.airflow.apache.org%3E

**Comment by ashb** [comment]
@mik-laj Do you have time to finish this off this week? We're getting close to cutting 2.1 branches, and I'd like to include this.

Or I can take it over if you don't have time.

**Comment by mik-laj** [comment]
@ashb you can go ahead and take over it.  I unfortunately have little time to contribute and focus only on review and support.

**Pull request closed without merging**